### ISSUE TITLE
mitmweb: version `ZLIB_1.2.9' not found

### ISSUE THREAD

**Issue by suntong** [mitmproxy/mitmproxy#4769]
#### Problem Description

    version `ZLIB_1.2.9' not found` 
error when starting `mitmweb`

It’s a bug of the binaries provided by mitmweb, which includes a number of libraries which are unpacked in a temporary directory at runtime.
The bundle of libraries includes libz version 1.2.8, which is too old to satisfy the requirements of libpng16-16.

#### Steps to reproduce the behavior:

1. start `mitmweb`
2. check the console print out:

```
$ mitmweb
/usr/bin/x-www-browser: /tmp/_MEIi9PAKI/libz.so.1: version `ZLIB_1.2.9' not found
 (required by /lib/x86_64-linux-gnu/libpng16.so.16)

$ apt-cache policy zlib1g-dev:amd64
zlib1g-dev:
  Installed: 1:1.2.11.dfsg-2
  Candidate: 1:1.2.11.dfsg-2
  Version table:
 *** 1:1.2.11.dfsg-2 500
        500 http://deb.debian.org/debian bullseye/main amd64 Packages
        100 /var/lib/dpkg/status

$ apt-cache policy libpng16-16:amd64
libpng16-16:
  Installed: 1.6.37-3
  Candidate: 1.6.37-3
  Version table:
 *** 1.6.37-3 500
        500 http://deb.debian.org/debian bullseye/main amd64 Packages
        100 /var/lib/dpkg/status


$ realpath /usr/bin/x-www-browser
/export/opt/google/chrome/google-chrome
```

#### System Information

```
$ mitmproxy --version
Mitmproxy: 7.0.2 binary
Python:    3.9.5
OpenSSL:   OpenSSL 1.1.1k  25 Mar 2021
Platform:  Linux-5.10.0-6-amd64-x86_64-with-glibc2.31

$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux bullseye/sid
```

**Comment by mhils** [comment]
Thanks for the report! What happens if you execute `x-www-browser` from the command line?

**Comment by suntong** [comment]
Executing `x-www-browser` from the command line on my system is just fine:

```
$ x-www-browser
Opening in existing browser session.
```

What happened was, `mitmweb` ships a number of libraries and unpacked them in a temporary directory at runtime; `LD_LIBRARY_PATH` is set to the temporary directory so that the dynamic linker will use the libraries there. This works fine for the binaries shipped by mitmweb, but `LD_LIBRARY_PATH` is passed down to the browser which mitmweb attempts to start, which in turn failed and gave the error message in OP.

Maybe unset  `LD_LIBRARY_PATH` when starting `x-www-browser`?

**Comment by mhils** [comment]
Yes, that analysis seems to be spot on. Right now we just use the `webbrowser` module from Python's standard library:

https://github.com/mitmproxy/mitmproxy/blob/d9d9a20ef2607f56c3e22114cc08d404930b0fc1/mitmproxy/tools/web/webaddons.py#L58-L66

See also https://github.com/pyinstaller/pyinstaller/issues/4835#issuecomment-678651604 and https://github.com/pyinstaller/pyinstaller/issues/3668#issuecomment-428951753

**Comment by milovangudelj** [comment]
Sorry for jumping in here like this, but how would a regular user that's just trying to use mitmweb on a linux machine go about solving the issue? 
I'm using it to try and understand how an app api works and I need the transparent mode that is not supported on windows.

**Comment by mhils** [comment]
The workaround for this is to run mitmweb with `--no-browser` and then start a browser manually. I'd be happy to merge a PR that fixes this issue properly. I can't reproduce it locally, which makes fixing tricky for me.

**Comment by milovangudelj** [comment]
Thank you! That worked perfectly. 
I used the command `./mitmweb --mode transparent --no-web-open-browser` and then navigated to the web server address manually.
